src/main/java/de/tum/in/www1/artemis/service/ExamService.java

```diff
@@ -573,11 +576,28 @@ public Integer startExercises(Long examId) {
                     }
                 }
             }
-        });
+        }));
 
         return generatedParticipations.size();
     }
 
+    private void executeInParallel(Runnable task) {
+        final int numberOfParallelThreads = 10;
+        ForkJoinPool forkJoinPool = new ForkJoinPool(numberOfParallelThreads);
+        Future<?> future = forkJoinPool.submit(task);
+        forkJoinPool.shutdown();
```

Comment:
@MaisiKoleni can you comment in the Code why the shutdown is necessary here? It looks wrong to me here

Reply:
This is also the way how it was in the first commit. But I agree that it looks out of place there, I change that.